builtin/credential/ldap/backend.go

```diff
@@ -122,6 +122,17 @@ func (b *backend) Login(req *logical.Request, username string, password string)
 		return nil, logical.ErrorResponse(fmt.Sprintf("LDAP bind failed: %v", err)), nil
 	}
 
+	// We re-bind to the BindDN if it's defined because we assume
+	// the BindDN should be the one to search, not the user logging in.
+	if cfg.DiscoverDN || (cfg.BindDN != "" && cfg.BindPassword != "") {
```

Comment:
@shomron, should I just get rid of `cfg.DiscoverDN` clause?  And simply re-bind if the second clause is true?  That makes the most sense to me.